core/src/main/java/org/apache/carbondata/core/util/QuadTreeUtil.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.core.util
+
+import java.util
+
+
+class QuadTreeUtil{
+
+  /**
+    * 二分查找法,传入的是列表  List<long[]> arrays, 待查询的是 des
+    *
+    * @param arrays 数据区域,数据是有序的
+    * @param des    待查询数据
+    * @return >=0 返回找到的元素在列表中的位置  < 0 未找到
+    */
+  def binarySearch(arrays: util.List[Array[Long]], des: Long): (Int,(Int, Int) ) = {
```

Comment:
Can we reuse the collections search implementation, why write our own binary search?

Reply:
This file is removed from this PR and it will be raised by @MarvinLitt  in a different PR as part algorithm.